src/Microsoft.Extensions.Logging.EventSource/EventSourceLoggerProvider.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Extensions.Logging.EventSourceLogger
+{
+    /// <summary>
+    /// The provider for the <see cref="EventSourceLogger"/>.
+    /// </summary>
+    internal class EventSourceLoggerProvider : ILoggerProvider
+    {
+        // A small integer that uniquely identifies the LoggerFactory assoicated with this LoggingProvider.
+        // Zero is illegal (it means we are uninitialized), and have to be added to the factory. 
+        private int _factoryID;
+
+        private readonly ILoggerFactory _loggerFactory;
```

Comment:
Can you be more specific?   Are you talking about the factoryID?     Can you explain your rationale for not needing it more?   It seems like EventListeners/ETW still have a model where there is only one logical instance of the logger, and thus if you are monitoring a scenario that has created several LoggerFactories (e.g. multi-tenant case), you still need to have the factory ID if you want to avoid ambiguity.